lib/cortex/snippets/client/helper.rb

```diff
@@ -70,7 +72,7 @@ def dynamic_yield
         private
 
         def webpage
-          Cortex::Snippets::Client::current_webpage(request)
+          Cortex::Snippets::Client::current_webpage(request, @cortex_client)
```

Comment:
There's no guarantee this instance variable will exist at runtime, since it's set by the `snippet` method, which could get called before or after something invokes `webpage`. Additionally, `@cortex_client` could get overwritten by something that re-invokes the `snippet` method, changing the value of `webpage` - this means `webpage` isn't thread-safe, which is problematic if we move to Puma in Employer. The solution here is to move away from the singleton pattern and turn this module into a class, and pass the `cortex_client` into the constructor.